### DISCUSSION TITLE
Overriding matchHosts username/password per repository

### DISCUSSION THREAD

**Discussion by adamj88** [renovatebot/renovate#21001] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.10.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

When providing a username and password for a host across a group of repositories, I would like to be able to override the username and password on a per-repository basis.

### What you tried so far.

I run a configuration against a group of repositories, when accessing a host and setting a username/password across all using hostRules the connection is fine, however, when attempting to override the username/password for a specific repository the original hostRules are used instead of accessing with a different username and password.

```
{
  ...
  "hostRules": [
    {
      "matchHost": "repo.host.com",
      "username": "username",
      "password": "password"
    }
  ],
  ...
  "repositories": [
    {
      "repository": "a/specific/repository",
      "hostRules": [
        {
          "matchHost": "repo.host.com",
          "username": "different-username",
          "password": "different-password"
        }
      ]
    }
  ]
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by adamj88** [comment]
I found the following to try and overcome this in the documentation which suggests a priority order, but this still yielded the same results:

> If multiple hostRules match a request, then they will be applied in the following order/priority:
> 
> - rules with only hostType specified
> - rules with only matchHost specified (sorted by matchHost length if multiple match)
> - rules with both matchHost and hostType specified (sorted by matchHost length if multiple match)

Changing my config to give the global rule a hostType to reduce its priority:

```
{
  ...
  "hostRules": [
    {
      "matchHost": "repo.host.com",
      "hostType": "packagist",
      "username": "username",
      "password": "password"
    }
  ],
  ...
  "repositories": [
    {
      "repository": "a/specific/repository",
      "hostRules": [
        {
          "matchHost": "repo.host.com",
          "username": "different-username",
          "password": "different-password"
        }
      ]
    }
  ]
}
```

**Reply by rarkins** [reply]
In the docs you quote, the 3rd in the list is the highest priority.
